src/views/table_cell.ts

```diff
@@ -0,0 +1,92 @@
+/*!
+ * © 2019 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { schema } from '@manuscripts/transform'
+import { DOMSerializer, ResolvedPos } from 'prosemirror-model'
+
+import {
+  ContextMenu,
+  ContextMenuButton,
+} from '../components/views/TableCellContextMenu'
+import BlockView from './block_view'
+import { createNodeView } from './creators'
+import { EditableBlockProps } from './editable_block'
+import ReactSubView from './ReactSubView'
+
+export class TableCellView extends BlockView<EditableBlockProps> {
+  public contentDOM: HTMLElement
+
+  public initialise = () => {
+    if (!this.node.type.spec.toDOM) {
+      return
+    }
+    const outputSpec = this.node.type.spec.toDOM(this.node)
+    const { dom } = DOMSerializer.renderSpec(document, outputSpec)
+    this.dom = dom as HTMLElement
+    this.contentDOM = document.createElement('span')
+    this.dom.appendChild(this.contentDOM)
+
+    if (isInTable(this.view.state.doc.resolve(this.getPos()))) {
```

Comment:
Why is this check there?

Reply:
this to prevent showing the menu for the first and last row in the table

Reply:
Why don't we want it in the first/last row?

Reply:
I derived that from the table actions in the [menu](https://github.com/Atypon-OpenSource/manuscripts-body-editor/blob/master/src/commands.ts#L772), I don't know why they disable them for that case. so we should show it for all table cells?

Reply:
That logic looks wrong to me. I'm guessing it's from the time where we had our own table schema. I think we should get rid of `isInTableBody` completely.

Reply:
great, updated